src/middlewared/middlewared/plugins/account_/2fa.py

```diff
@@ -0,0 +1,45 @@
+import pyotp
+
+from middlewared.schema import accepts, Bool, Int, returns, Str
+from middlewared.service import CallError, Service
+
+
+class UserService(Service):
+
+    @accepts(Int('user_id'))
+    @returns(Str(title='Provisioning URI'))
+    async def provisioning_uri(self, user_id):
```

Comment:
Is there any way we can use uid values rather than DB IDs? Former are easier to get for AD / LDAP users (since they don't exist in our sqlite database).

Reply:
I believe those changes we can account for when we add AD support ? Motivation is that we can comprehensively discuss how to go about AD and make the changes. These changes are not stable yet and we can get started with AD right away ?

Reply:
By stable i mean they are not going in a stable release right now :)